cmd/storj/cmd/cp.go

```diff
@@ -32,79 +37,164 @@ func init() {
 	cpCmd.Flags().String("config", filepath.Join(defaultConfDir, "config.yaml"), "path to configuration")
 }
 
-func copy(cmd *cobra.Command, args []string) (err error) {
-	ctx := process.Ctx(cmd)
+// upload uploads args[0] from local machine to s3 compatible object args[1]
+func upload(ctx context.Context, bs buckets.Store, srcFile string, destObj *url.URL) error {
+	if destObj.Scheme == "" {
+		fmt.Println("Invalid destination")
+		return nil
+	}
 
-	if len(args) == 0 {
-		return errs.New("No file specified for copy")
+	// if object name not specified, default to filename
+	if destObj.Path == "" || destObj.Path == "/" {
+		destObj.Path = filepath.Base(srcFile)
 	}
 
-	if len(args) == 1 {
-		return errs.New("No destination specified")
+	f, err := os.Open(srcFile)
+	if err != nil {
+		return err
 	}
 
-	so, err := getStorjObjects(ctx, cpCfg)
+	defer utils.LogClose(f)
+
+	o, err := bs.GetObjectStore(ctx, destObj.Host)
 	if err != nil {
 		return err
 	}
 
-	u, err := url.Parse(args[0])
+	meta := objects.SerializableMeta{}
+	expTime := time.Time{}
+
+	_, err = o.Put(ctx, paths.New(destObj.Path), f, meta, expTime)
 	if err != nil {
 		return err
 	}
 
-	if u.Scheme == "" {
-		f, err := os.Open(args[0])
-		if err != nil {
-			return err
-		}
+	fmt.Printf("Created: %s\n", destObj.Path)
 
-		fi, err := f.Stat()
-		if err != nil {
-			return err
-		}
+	return nil
+}
 
-		fr, err := hash.NewReader(f, fi.Size(), "", "")
-		if err != nil {
-			return err
-		}
+// download downloads s3 compatible object args[0] to args[1] on local machine
+func download(ctx context.Context, bs buckets.Store, srcObj *url.URL, destFile string) error {
```

Comment:
maybe a srcObj scheme check like in upload?

Reply:
Done